### ISSUE TITLE
Release 2.0.0

### ISSUE THREAD

**Issue by tobiasKaminsky** [nextcloud/android#1304]
- [x] Bump version name and version code in **all (3)** ```AndroidManifest.xml``` files
- [x] Update ```CHANGELOG.md``` adding new features
- [x] Add #1282 and test it
- [ ] https://github.com/nextcloud/android/issues/1081
- [ ] update translations for Fdroid & Google Play Store

**Test app with all supported Android version**
- access all activities
- auto upload
- upload / download files/folders
- use external sdcard
- what's new slides correctly shown on first start

All tests on
  - [x] Android 4
  - [x] Android 5
  - [x] Android 6
  - [x] Android 7

**RC2**
- [x] Verify that the new version number is shown in settings
- [x] Tag new version as ```rc-2.0.0-02```
- [x] Build and release to Play Store as beta release and :shipit: 
- [x] ~Publish it on f-droid as optional release: edit metadata and open new merge request on gitlab~
- [x] ~Verify that it is shown on f-droid (within 24h)~ (Currently not possible due to toolchain)

**RC3**
- [x] Verify that the new version number is shown in settings
- [x] Tag new version as ```rc-2.0.0-03```
- [x] Build and release to Play Store as beta release and :shipit: 
- [x] ~Publish it on f-droid as optional release: edit metadata and open new merge request on gitlab~
- [x] ~Verify that it is shown on f-droid (within 24h)~ (Currently not possible due to toolchain)

**RC4**
- [x] Verify that the new version number is shown in settings
- [x] Tag new version as ```rc-2.0.0-04```
- [x] Build and release to Play Store as beta release and :shipit: 
- [x] ~Publish it on f-droid as optional release: edit metadata and open new merge request on gitlab~
- [x] ~Verify that it is shown on f-droid (within 24h)~ (Currently not possible due to toolchain)

**RC5**
- [x] Verify that the new version number is shown in settings
- [x] Tag new version as ```rc-2.0.0-05```
- [x] Build and release to Play Store as beta release and :shipit: 
- [x] Publish it on f-droid as optional release: edit metadata and open new merge request on gitlab
- [x] Verify that it is shown on f-droid (within 24h)

**RC6**
- [x] Verify that the new version number is shown in settings
- [x] Tag new version as ```rc-2.0.0-06```
- [x] Build and release to Play Store as beta release and :shipit: 
- [x] Publish it on f-droid as optional release: edit metadata and open new merge request on gitlab
- [x] Verify that it is shown on f-droid (within 24h)

**RC7**
- [x] Verify that the new version number is shown in settings
- [x] Tag new version as ```rc-2.0.0-07```
- [x] Build and release to Play Store as beta release and :shipit: 
- [x] Publish it on f-droid as optional release: edit metadata and open new merge request on gitlab
- [x] Verify that it is shown on f-droid (within 24h)

**RC8**
- [x] Verify that the new version number is shown in settings
- [x] Tag new version as ```rc-2.0.0-08```
- [x] Build and release to Play Store as beta release and :shipit: 
- [x] Publish it on f-droid as optional release: edit metadata and open new merge request on gitlab
- [ ] Verify that it is shown on f-droid (within 24h)

**RC9**
- [x] Verify that the new version number is shown in settings
- [x] Tag new version as ```rc-2.0.0-09```
- [x] Build and release to Play Store as beta release and :shipit: 
- [x] Publish it on f-droid as optional release: edit metadata and open new merge request on gitlab
- [ ] Verify that it is shown on f-droid (within 24h)

**Final Release, after two weeks time (max) of latest RC release**
- [ ]  replace screenshots (https://github.com/nextcloud/screenshots/tree/master/android/phone)
- [x] Verify that the new version number is shown in settings
- [x] Verify that logging is disabled
- [x] Tag new version as ```stable-2.0.0```
- [x] Build and release to Play Store as stable release and :shipit: 
- [x] Update Play store changelog based on condensed ```CHANGELOG.md``` infos
- [x] Verify that it is shown on f-droid (within 24h) and close #1483
- [ ] Publish APK to https://download.nextcloud.com/android/

**Comment by tobiasKaminsky** [comment]
Although google play console shows still some bugs, I suggest that we release the RC5 as a final within the next days.
Then we can focus on bug fixing for a 2.0.1 and in parallel work on a new feature release 2.1 in parallel as @mario suggested with the "new" git workflow.

What do you think? @mario @AndyScherzinger

**Comment by TFLye** [comment]
nice  idea

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky In general I'd say yes, but two issues still seem to be critical in my opinion (can't say if they are 2.0.0 only though):

* #1450
* #1419 (might be fixed with RC5)

**Comment by mario** [comment]
@tobiasKaminsky what's the progress on moving all our discussions/ideas to github (from your notebook)? :)

There are quite a few bugs reported (some on Github, some on Play Services, some on Forums) so I'd really opt to fix critical bugs before we release since this is going to be a BIG release, should NOT screw it up.

**Comment by tobiasKaminsky** [comment]
Transferring from notebook will be next week, currently I am busy on the other project :/
I am trying to fix some bugs now.

**Comment by tobiasKaminsky** [comment]
@Bubu do you have time to create push RC6 on fdroid? Would be very great :tada:

**Comment by Bubu** [comment]
@tobiasKaminsky Will do!

**Comment by tobiasKaminsky** [comment]
Outstanding issues/prs for RC 7:
- [ ] https://github.com/nextcloud/android/pull/1539
- [ ] https://github.com/nextcloud/android/pull/1560
- [ ] https://github.com/nextcloud/android-library/pull/90
- [ ] new lib release & integrate in master
- [ ] https://github.com/nextcloud/android/issues/1081

All other bugs are either "need info" or can imho postponed to 2.0.1.
So after finishing this list I would like to do a RC7 and after that the final 2.0.0

@mario @AndyScherzinger

**Comment by AndyScherzinger** [comment]
Perfectly fine with we 👍

**Comment by tobiasKaminsky** [comment]
@Bubu can you do another RC release for us on fdroid? RC7 with tag c-2.0.0-07 (https://github.com/nextcloud/android/releases/tag/rc-2.0.0-07)
Thank you :tada:

**Comment by Bubu** [comment]
@tobiasKaminsky Done: https://gitlab.com/fdroid/fdroiddata/commit/9ec1d7e6a410280335317caffe179b9ab64f7ebf

**Comment by enoch85** [comment]
Just a quick question, how do I leave the beta program on Google Play once 2.0.0 is released?

**Comment by mario** [comment]
You'll be automatically upgraded to 2.0.0 stable, but to leave completely:

https://play.google.com/apps/testing/com.nextcloud.client

**Comment by enoch85** [comment]
Thanks @mario :)

**Comment by tobiasKaminsky** [comment]
@Bubu may I ask you for the next RC? Hopefully RC8 is the last one *finger-crossing*

**Comment by Bubu** [comment]
@tobiasKaminsky Done. The latest stable version was moved to the archive after the last RC (because default is to keep only 3 versions.). I fixed this as well, so RC8 and latest 1.x will be available (again) in 1-2 days.

**Comment by tobiasKaminsky** [comment]
@Bubu can you do another (hopefully last) RC9?
https://github.com/nextcloud/android/releases/tag/rc-2.0.0-09

Regarding the app versions, I guess the best is to only show the latest RC and at least the latest 2-3 stable version, in case someone wants to downgrade.
RC can then be disabled if version is stable.

Thank you 🥇

**Comment by Bubu** [comment]
@tobiasKaminsky Done. For showing only latest RC fdroid is lacking the functionality to manually mark versions as archived. They can only be deleted but I havn't done this for now.

**Comment by tobiasKaminsky** [comment]
You are the expert on fdroid, but I thought that we could "just" put disable on certain build receipts?
https://f-droid.org/docs/Build_Metadata_Reference/#Build

**Comment by Bubu** [comment]
This works, but completely removes those builds from the server. I would prefer to just move those obsolete RCs to the archive repo along with all the other old versions. I'll ask what the others think about this, but I guess it's not that important to keep old RC's archived for now.

**Comment by tobiasKaminsky** [comment]
Old RCs are indeed, at least for me, not important. 
Each major/minor release of course should still be available, at least as a build receipt.

**Comment by Bubu** [comment]
Okay, I disabled RC5-7 (fdroid didn't have the ones before that). Will disable RC8 and RC9 after 2.0 is published.

**Comment by mario** [comment]
Thanks, you're awesome!

**Comment by AndyScherzinger** [comment]
Thanks @Bubu! 🎉

**Comment by AndyScherzinger** [comment]
@Bubu 2.0.0 final has been tagged :)

**Comment by Bubu** [comment]
It's been picked up by the f-droid bot already, let's see when it will be build! :)

**Comment by Bubu** [comment]
> Will disable RC8 and RC9 after 2.0 is published.

Done now, also I reset the number of available versions in the main repo to 3. Everything older than that is available form the archive.
